plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/explorer/OpenShiftExplorerContentProvider.java

```diff
@@ -44,197 +41,157 @@
  * 
  * @author jeff.cantrill
  */
-public class OpenShiftExplorerContentProvider extends BaseExplorerContentProvider 
-	implements PropertyChangeListener, IProjectCache.IProjectCacheListener {
-	
+public class OpenShiftExplorerContentProvider implements ITreeContentProvider {
 	private static final List<String> TERMINATED_STATUS = Arrays.asList("Complete", "Failed", "Error", "Cancelled");
-	private static final IProjectCache cache = new OpenShiftProjectCache();
-	
+
+	private OpenshiftUIModel model;
+	private IElementListener listener;
+	private StructuredViewer viewer;
+	private Map<Object, BaseExplorerContentProvider.LoadingStub> stubs= new HashMap<Object, BaseExplorerContentProvider.LoadingStub>();
+
 	public OpenShiftExplorerContentProvider() {
-		super();
-		cache.addListener(this);
+		this(new OpenshiftUIModel());
```

Comment:
Why OpenshiftUIModel remains hidden inside OpenShiftExplorerContentProvider. Can it be of use for other UI components/actions? Let's have a singleton instance for it.